pkg/cmd/cli/cmd/newapp.go

```diff
@@ -271,15 +274,56 @@ func setLabels(labels map[string]string, result *newcmd.AppResult) error {
 	return nil
 }
 
+// isInvalidTriggerError returns true if the given error is
+// a validation error that contains 'trigger' in its error message.
+// This error is returned from an older server when
+// it doesn't recognize a newer trigger type.
+func isInvalidTriggerError(err error) bool {
+	if !kapierrors.IsInvalid(err) {
+		return false
+	}
+	statusErr, ok := err.(*kapierrors.StatusError)
+	if !ok {
+		return false
+	}
+	return strings.Contains(statusErr.Status().Message, "trigger")
+}
+
+// retryBuildConfig determines if the given error is caused by an invalid trigger
+// error on a BuildConfig. If that is the case, it will remove all triggers with a
+// type that is not in the whitelist for an older server.
+func retryBuildConfig(info *resource.Info, err error) runtime.Object {
+	triggerTypeWhiteList := map[buildapi.BuildTriggerType]struct{}{
+		buildapi.GitHubWebHookBuildTriggerType:  struct{}{},
+		buildapi.GenericWebHookBuildTriggerType: struct{}{},
+		buildapi.ImageChangeBuildTriggerType:    struct{}{},
```

Comment:
> I think that's fine, our code ignores that trigger and there's no way to have nil pointer access on that, since we don't know about Foo and we'll never try to read fooTrigger.

What about when the server gets upgraded to a version that knows about type=Foo and expects a non-nil FooParams (and has validation in place that ensures non-nil params, but that validation wasn't run when the object was created against the old API)